shared/src/business/entities/trialSessions/TrialSession.test.js

```diff
@@ -497,4 +585,213 @@ describe('TrialSession entity', () => {
       expect(trialSession.noticeIssuedDate).toBeTruthy();
     });
   });
+
+  describe('proceedingType', () => {
+    it('should throw an error when passed an invalid proceedingType', () => {
+      const trialSession = new TrialSession(
+        {
+          ...VALID_TRIAL_SESSION,
+          address1: '123 Flavor Ave',
+          city: 'Flavortown',
+          judge: {},
+          postalCode: '12345',
+          proceedingType: 'NOT A VALID TYPE',
+          state: 'TN',
+        },
+        {
+          applicationContext,
+        },
+      );
+
+      expect(trialSession.getFormattedValidationErrors()).toMatchObject({
+        proceedingType: TrialSession.VALIDATION_ERROR_MESSAGES.proceedingType,
+      });
+    });
+
+    it('should be valid with a "Remote" proceedingType', () => {
+      const trialSession = new TrialSession(
+        {
+          ...VALID_TRIAL_SESSION,
+          address1: '123 Flavor Ave',
+          city: 'Flavortown',
+          judge: {},
+          postalCode: '12345',
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.remote,
+          state: 'TN',
+        },
+        {
+          applicationContext,
+        },
+      );
+
+      expect(trialSession.isValid()).toBeTruthy();
+    });
+
+    it('should be valid with an "In Person" proceedingType', () => {
+      const trialSession = new TrialSession(
+        {
+          ...VALID_TRIAL_SESSION,
+          address1: '123 Flavor Ave',
+          city: 'Flavortown',
+          judge: {},
+          postalCode: '12345',
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+          state: 'TN',
+        },
+        {
+          applicationContext,
+        },
+      );
+
+      expect(trialSession.isValid()).toBeTruthy();
+    });
+
+    it('should be invalid with no proceedingType', () => {
+      const trialSession = new TrialSession(
+        {
+          ...VALID_TRIAL_SESSION,
+          address1: '123 Flavor Ave',
+          city: 'Flavortown',
+          judge: {},
+          postalCode: '12345',
+          proceedingType: null,
+          state: 'TN',
+        },
+        {
+          applicationContext,
+        },
+      );
+
+      expect(trialSession.getFormattedValidationErrors()).toMatchObject({
+        proceedingType: TrialSession.VALIDATION_ERROR_MESSAGES.proceedingType,
+      });
+    });
+  });
+
+  describe('required fields when calendared', () => {
+    describe('proceedingType In Person', () => {
+      it('should be valid when isCalendared is true, proceedingType is In Person, and required address fields are missing', () => {
+        const trialSession = new TrialSession(
+          {
+            ...VALID_TRIAL_SESSION,
+            address1: undefined,
```

Comment:
Am I interpreting this correctly, that we don't want validation errors for missing required fields?

Reply:
In-person trial session support hasn't been added yet, because all sessions are currently remote.